### PR TITLE
Added Apptainer support

### PR DESCRIPTION
**Pull request by mambelli**
Internal variables are still named ...SINGULARITY... but reading and writing APPTAINER_ variables as well

GlideinWMS is not reading SINGULARITY_ environment variables. It is setting one
All singularity used vars:
   2 SINGULARITY_BINDPATH
Only in comments, GWMS is not checking or setting it
   1 SINGULARITY_HOME
Only in comments, GWMS is not checking or setting it
   4 SINGULARITY_NAME
Only in comments, GWMS is not checking or setting it
  11 SINGULARITY_WORKDIR
Only set, setting also APPTAINER_WORKDIR and updated BATS test

GlideinWMS is reading, clearing, or setting SINGULARITYENV_ variables. Treating the same also APPTAINERENV_

GlideinWMS is parsing and interpreting correctly also the apptainer version, a new variable GWMS_CONTAINERSW_FULL_VERSION, allows to distinguish between singularity and apptainer.

GlideinWMS is looking also for the "apptainer" binary (currently not needed since Apptainer is committed to providing a singularity symlink)

This PR fixes also a bug with the BATS test runner that was not detecting correctly some failures (separate commit).

### CODE REVIEW

**Review by namrathaurs — APPROVED** [review]
Changes look good and can be merged.

**Comment by mambelli** [comment]
Improved the code detecting the sw version correctly and looking also for the apptainer executable

**Review by namrathaurs — COMMENTED** [review]
Added a clarification comment for the apptainer support workflow with respect to the checking order implemented in relation to the expected preference ('apptainer' first and then 'singularity')

File: creation/web_base/singularity_lib.sh [hunk 1]
```diff
@@ -1356,19 +1373,20 @@ singularity_test_bin() {
...
     # Find Singularity in search path, check the version and validate w/ the image (if an image is passed)
+    # Will look in some places also for apptainer (and singularitypro)
     # This will search in order:
     # 1. Optional: Look first in the override path (GLIDEIN_SINGULARITY_BINARY_OVERRIDE)
-    # 2. Look in the path suggested via $1 (SINGULARITY_BIN)
+    # 2. Look for 'singularity' and then 'apptainer' in the path suggested via $1 (SINGULARITY_BIN)
     #      (keywords: PATH -> go to step 3 - ie start w/ $PATH;
     #           OSG -> OSG location, and continue from step 3 if failed, this is the default)
-    # 3. Look in $PATH
+    # 3. Look in $PATH for 'singularity' and then 'apptainer'
```
[the remark this answers is not in the document]
Comment (mambelli) [reply creation/web_base/singularity_lib.sh]: Order changed.

**Review by namrathaurs — APPROVED** [review]
Requested changes look fine. Ready for merging!

**Pull request merged**